src/audio/component.c

```diff
@@ -32,22 +32,21 @@ DECLARE_TR_CTX(comp_tr, SOF_UUID(comp_uuid), LOG_LEVEL_INFO);
 int comp_register(struct comp_driver_info *drv)
 {
 	struct comp_driver_list *drivers = comp_drivers_get();
-	uint32_t flags;
 
-	irq_local_disable(flags);
+	spin_lock(&drivers->lock);
 	list_item_prepend(&drv->list, &drivers->list);
-	irq_local_enable(flags);
+	spin_unlock(&drivers->lock);
```

Comment:
Does this mean, that this code had two bugs: firstly it was needlessly disabling interrupts, and secondly there was a race on SMP because disabling local interrupts doesn't protect against concurrent access from multiple cores? Calls to `comp_register()` are only done on core 0 during initialisation. `comp_unregister()` is actually never used so far. As for `get_drv()` below it's only called from `comp_new()` which also IPC-serialised? So is any protection needed at all?

Reply:
No protection needed today, but what happens when we can load/unload modules - then the lock is needed.